Dockerfile

```diff
@@ -0,0 +1,34 @@
+FROM rust:1.67-bullseye AS chef
```

Comment:
what does the chef step do? I'm not familiar with this

Reply:
Forgot to explain this bit. With how the Rust build system works, Docker doesn't quite understand how the dependencies are built, and thus will force you to rebuild all your dependencies (and transitive deps) on every Docker build. `cargo-chef` (https://github.com/LukeMathWalker/cargo-chef) is a Cargo extension that splits out building your dependencies into a separate step that Docker can track and cache.

So, provided you don't make any changes to your Cargo.lock, you should only have to rebuild your immediate crate instead of all your dependencies on every Docker build

Reply:
oh that's cool, thanks for the info